financial-connections/src/main/java/com/stripe/android/financialconnections/features/linkaccountpicker/LinkAccountPickerViewModel.kt

```diff
@@ -255,14 +349,33 @@ internal data class LinkAccountPickerState(
         val consumerSessionClientSecret: String,
         val defaultCta: String,
         val nextPaneOnNewAccount: Pane?,
-        val partnerToCoreAuths: Map<String, String>?
-    )
-
-    val cta: String?
-        get() = payload()?.let { payload ->
-            payload.accounts.firstOrNull { it.first.id == selectedAccountId }
-                ?.second?.selectionCta
-                ?: payload.defaultCta
+        val partnerToCoreAuths: Map<String, String>?,
+        val singleAccount: Boolean,
+    ) {
+
+        fun selectedPartnerAccounts(selected: List<String>): List<PartnerAccount> {
+            return accounts.filter { it.first.id in selected }.map { it.first }
+        }
+    }
+
+    val cta: TextResource
+        get() {
+            val payload = payload()
+
+            return if (payload?.singleAccount == true) {
```

Comment:
Updated logic for the button label:
- For single accounts, use the selected account or the default label.
- For multiple accounts, use the singular/plural based on the selected count.

(cc @kgaidis-stripe)

Reply:
I believe web uses cta from the backend (default_cta) not a client side one?

https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/ebf95015/src/linkedAccounts/inner/components/panes/v3/LinkAccountPickerPane/containers/ReturningUser.tsx#L271

for single account, I do modify the CTA based off "selected account" (or default label)

for multi, I go to default

web just always go to default? (which is likely mostly fine anyway, even if not 100% correct because update required drawer covers it)

Reply:
I’d love to be able to use singular/plural, but I guess I’ll also switch to what we do on Web!